pulsar-flink-connector/src/test/java/org/apache/flink/streaming/connectors/pulsar/RowDataDeSerializationSchemaTest.java

```diff
@@ -0,0 +1,369 @@
+
+    @Test
+    public void testAvroSerializeDeserialize() throws Exception {
+        String topicName = newTopic();
+        final DataType dataType = getDataType();
+        final RowType rowType = (RowType) dataType.getLogicalType();
+        final TypeInformation<RowData> typeInfo = InternalTypeInfo.of(rowType);
+
+        Schema avroSchema = AvroSchemaConverter.convertToSchema(rowType);
+        if (avroSchema.isNullable()) {
+            avroSchema = avroSchema.getTypes().stream().filter(s -> s.getType() == RECORD).findAny().get();
+        }
+        final GenericRecord record = getGenericRecord(avroSchema);
+        byte[] input = buildGenericRecordBytes(avroSchema, record);
+
+        AvroRowDataSerializationSchema serializationSchema = new AvroRowDataSerializationSchema(rowType);
+        serializationSchema.open(null);
+        AvroRowDataDeserializationSchema deserializationSchema =
+                new AvroRowDataDeserializationSchema(rowType, typeInfo);
+        deserializationSchema.open(null);
+
+        RowData rowData = deserializationSchema.deserialize(input);
+        byte[] output = serializationSchema.serialize(rowData);
+        final org.apache.pulsar.client.api.Schema<RowData> pulsarSchema =
+                toPulsarSchema(SchemaType.AVRO, avroSchema,
+                        serializationSchema, deserializationSchema);
+        sendMessage(topicName, pulsarSchema, output);
+        final CompletableFuture<byte[]> consumer = autoConsumer(topicName
+        );
+
+        RowData newRowData = deserializationSchema.deserialize(consumer.get(2000, TimeUnit.MILLISECONDS));
+        assertEquals(rowData, newRowData);
+        assertArrayEquals(input, output);
+    }
+
+    @Test
+    public void testJsonSerializeDeserialize() throws Exception {
+        String topicName = newTopic();
+        DataType dataType = getJsonDataType();
+        final RowType rowType = (RowType) dataType.getLogicalType();
+        final TypeInformation<RowData> typeInfo = InternalTypeInfo.of(rowType);
+
+        Schema avroSchema = AvroSchemaConverter.convertToSchema(rowType);
+        if (avroSchema.isNullable()) {
+            avroSchema = avroSchema.getTypes().stream().filter(s -> s.getType() == RECORD).findAny().get();
+        }
+        byte[] serializedJson = getJsonBytes();
+
+        JsonRowDataSerializationSchema serializationSchema =
+                new JsonRowDataSerializationSchema(rowType, TimestampFormat.ISO_8601, JsonOptions.MapNullKeyMode.DROP,
+                        "");
+        serializationSchema.open(null);
+        JsonRowDataDeserializationSchema deserializationSchema = new JsonRowDataDeserializationSchema(rowType, typeInfo,
+                false, false, TimestampFormat.ISO_8601);
+        deserializationSchema.open(null);
+
+        RowData rowData = deserializationSchema.deserialize(serializedJson);
+        byte[] output = serializationSchema.serialize(rowData);
+        assertArrayEquals(serializedJson, output);
```

Comment:
This assert doesn't seem to work for pulsar validation, but for jsonSer/Deser validation.